sessions/2018/july/Session_9.md

```diff
@@ -0,0 +1,124 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 9 (07/08/2018)
+
+- TDD
+
+### Notes:
+
+#### TDD
+
+There exists 3 types of Test framework (terms can vary): *Unit*, *component* and *integration*.
+
+##### Unit Tests
+Unit tests are used for only test a single class.
+
+##### Component Tests
+Component tests are very fast and are intended for multiple clases but not for I/O or databases. A common pattern of this is to test a package.
+
+##### Integration Tests
+Integration tests are used to test classes and databases. In this category we can found two tendencies:
```

Comment:
Not testing databases themselves, but testing that classes that interact with databases work as intended.